edb/schema/types.py

```diff
@@ -1937,36 +1979,80 @@ def generate_array_type_id(
     return generate_type_id(id_basis)
 
 
-def get_union_type_id(
+def get_union_type_id_and_name(
```

Comment:
Why do the compound/collection types need to have deterministically computed ids? Why don't the names work to do shell resolution?

(Document whatever the answer is here, since I'm probably missing something)

Reply:
Hm. Good point. I'll check if id determinism is still useful.

Reply:
Yeah, stable ids aren't needed anymore